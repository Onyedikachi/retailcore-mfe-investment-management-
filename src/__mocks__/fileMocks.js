module.exports = "";
export let ProviderValue = {
  isChecker: false,
  selected: {
    id: 1,
    text: "created by me",
    value: "created by me",
  },
  setSelected: jest.fn(),
  setIsChecker: jest.fn(),
  category: "all branches",
  setCategory: jest.fn(),
  setStatus: jest.fn(),
  status: "",
  dateData: "",
  setDateData: jest.fn(),
  search: "",
  setSearch: jest.fn(),
  type: "",
  setType: jest.fn(),
  initiator: "",
  setInitiator: jest.fn(),
  setDuration: jest.fn(),
  duration: "",
  isRefreshing: false,
  setRefreshing: jest.fn(),
};
