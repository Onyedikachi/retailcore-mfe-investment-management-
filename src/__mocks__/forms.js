export const countries = [
  {
    id: 1,
    name: "Nigeria",
    value: "Nigeria",
  },
  {
    id: 2,
    name: "Ghana",
    value: "Ghana",
  },
];

export const states = [
  {
    id: 1,
    name: "Lagos",
    value: "Lagos",
  },
  {
    id: 2,
    name: "Ibadan",
    value: "Ibadan",
  },
];

export const LGAs = [
  {
    id: 1,
    name: "Abuja",
    value: "Abuja",
  },
  {
    id: 2,
    name: "Asokoro",
    value: "Asokoro",
  },
];
