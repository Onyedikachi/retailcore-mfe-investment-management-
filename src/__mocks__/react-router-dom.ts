export const useParams = jest.fn();
export const useLocation = jest.fn();
