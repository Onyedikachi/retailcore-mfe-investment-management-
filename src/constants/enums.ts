export enum StatusCategoryType {
  AllBranches = "all branches",
  Requests = "requests",
  // Add other category types here
}

export enum StatusItemType {
  Type = "type",
  Request = "request",
}
export enum RequestStatus {
  APPROVED = "Approved",
  INREVIEW = "In-Review",
  INISSUE = "In-Issue",
  DRAFT = "Draft",
}

export enum ProductStatus {
  ACTIVE = "Active",
  INACTIVE = "InActive",
}

export enum Messages {
  ADMIN_BRANCH_SETUP_SENT = "Branch Setup Sent for Approval",
  BRANCH_CREATE_SUCCESS = "Branch created successfully",
  BRANCH_CREATE_FAILED = "Branch creation failed",
  BRANCH_CREATE_APPROVED = "Branch creation approved",
  BRANCH_CREATE_REJECTED = "Branch creation rejected",
  BRANCH_BULK_CREATE_APPROVED = "Branch bulk creation approved",
  BRANCH_BULK_CREATE_REJECTED = "Branch bulk creation rejected",
  ADMIN_BRANCH_CREATE_SUCCESS = "Branch creation request submitted for approval",
  ADMIN_BRANCH_CREATE_FAILED = "Branch creation request failed",
  BRANCH_MODIFY_SUCCESS = "Branch modified successfully",
  BRANCH_MODIFY_FAILED = "Branch modification failed",
  BRANCH_MODIFY_APPROVED = "Branch modification approved",
  BRANCH_MODIFY_REJECTED = "Branch modification rejected",
  ADMIN_BRANCH_MODIFY_SUCCESS = "Branch modification request submitted for approval",
  ADMIN_BRANCH_MODIFY_FAILED = "Branch modification request failed",
  BRANCH_ACTIVATE_SUCCESS = "Branch activated successfully",
  BRANCH_ACTIVATE_FAILED = "Branch activation failed",
  BRANCH_ACTIVATE_APPROVED = "Branch activation approved",
  BRANCH_ACTIVATE_REJECTED = "Branch activation rejected",
  ADMIN_BRANCH_ACTIVATE_SUCCESS = "Branch activation request submitted for approval",
  ADMIN_BRANCH_ACTIVATE_FAILED = "Branch activation request failed",
  BRANCH_DEACTIVATE_SUCCESS = "Branch deactivated successfully",
  BRANCH_DEACTIVATE_FAILED = "Branch deactivation failed",
  BRANCH_DEACTIVATE_APPROVED = "Branch deactivation approved",
  BRANCH_DEACTIVATE_REJECTED = "Branch deactivation rejected",
  ADMIN_BRANCH_DEACTIVATE_SUCCESS = "Branch deactivation request submitted for approval",
  ADMIN_BRANCH_DEACTIVATE_FAILED = "Branch deactivation request failed",
  BRANCH_WITHDRAW_DELETE_SUCCESS = "Request withdrawn and deleted successfully",
  BRANCH_WITHDRAW_DELETE_FAILED = "Unable to withdraw & delete",
  BRANCH_DELETE_SUCCESS = "Request deleted successfully",
  BRANCH_DELETE_FAILED = "Unable to delete",
  BRANCH_APPROVE_FAILED = "Unable to approve branch ",
  BRANCH_REJECT_FAILED = "Unable to reject branch ",
  BRANCH_DRAFT_SUCCESS = "Branch Setup saved to draft",
  BRANCH_DRAFT_FAILED = "Unable to save as draft",
}

export enum Prompts {
  BRANCH_DRAFT = "Do you want to save as draft?",
  BRANCH_DRAFT_SUBTEXT = "Requests in draft will be deleted after 2 hours of inactivity",
  BRANCH_ACTIVATE = "Do you want to activate this branch?",
  BRANCH_DEACTIVATE = "Do you want to deactivate this branch?",
  BRANCH_MODIFY = "Do you want to modify this branch?",
  REMOVE_RECORD = "Do you want to remove this record?",
  CANCEL_PROCESS = "Do you want to cancel process?",
  CANCEL_BRANCH_SETUP = "Do you want to cancel branch setup?",
  BRANCH_CREATION_REJECT = "Do you want to reject branch creation?",
  BRANCH_CREATION_APPROVE = "Do you want to approve branch creation?",
  BRANCH_BULK_CREATION_REJECT = "Do you want to reject bulk branch creation?",
  BRANCH_BULK_CREATION_APPROVE = "Do you want to approve bulk branch creation?",
  BRANCH_MODIFY_REJECT = "Do you want to reject branch modification?",
  BRANCH_MODIFY_APPROVE = "Do you want to approve branch modification?",
  BRANCH_MODIFY_CANCEL = "Do you want to cancel branch modification?",
  BRANCH_WITHDRAW_DELETE = "Do you want to withdraw and delete this request?",
  BRANCH_WITHDRAW_MODIFY = "Do you want to withdraw and modify this request?",
  BRANCH_DELETE = "Do you want to delete this request?",
  BRANCH_DEACTIVATION_APPROVE = "Do you want to approve branch deactivation",
  BRANCH_DEACTIVATION_REJECT = "Do you want to reject branch deactivation",
  BRANCH_ACTIVATION_APPROVE = "Do you want to approve branch activation",
  BRANCH_ACTIVATION_REJECT = "Do you want to reject branch activation",
}
