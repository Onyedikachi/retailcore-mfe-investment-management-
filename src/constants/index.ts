export * from "./permissions";
export * from "./branch";
export * from "./tables";
export * from "./form";
