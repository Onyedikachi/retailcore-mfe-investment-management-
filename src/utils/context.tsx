import { createContext } from "react";

interface ContextProps {
  category?: any;
  setCategory?: (e: any) => void;
  selected?: { id: number; text: string; value?: string } | null;
  setSelected?: (e: any) => void;
  isChecker?: boolean;
  setIsChecker?: (e: any) => void;
  setStatus?: (e: any) => void;
  status?: string;
  dateData?: any;
  setDateData?: (e: any) => void;
  search?: string;
  setSearch?: (e: any) => void;
  type?: string;
  setType?: (e: any) => void;
  initiator?: string;
  setInitiator?: (e: any) => void;
  setDuration?: (e: any) => void;
  duration?: string;
  isRefreshing?: boolean;
  setRefreshing?: (e: boolean) => void;
  role?: string;
  setRole?: (e: string) => void;
}

interface branchDeactivationContextProps {
  stage: number;
  setStage: (e: number) => void;
}

interface ISummaryContextProps {
  // branchesList: Array<any>;
  branch?: any;
  bulkBranchData?: any;
  setBulkBranchData?: (e: any) => void;
  requestData?: any;
  bulkQueryFetching?: boolean;
  formData?: any;
  uploadData?: any[];
}

export const AppContext = createContext<{
  role: string;
  setRole: (e) => void;
  permissions: string[];
}>({
  role: "",
  setRole: () => {},
  permissions: [],
});

export const InvestmentContext = createContext<ContextProps>({
  category: null,
  setCategory: () => {},
  selected: null,
  setSelected: () => {},
  isChecker: true,
  setIsChecker: () => {},
  setStatus: () => {},
  status: "",
  dateData: null,
  setDateData: () => {},
  search: "",
  setSearch: () => {},
  type: "",
  setType: () => {},
  initiator: "",
  setInitiator: () => {},
  setDuration: () => {},
  duration: "",
  isRefreshing: false,
  setRefreshing: () => {},
  role: "",
  setRole: () => {},
});

export const BranchDeactivationContext =
  createContext<branchDeactivationContextProps>({
    stage: null,
    setStage: () => {},
  });

export const SummaryContextProps = createContext<ISummaryContextProps>({
  branch: null,
  setBulkBranchData: () => {},
  requestData: null,
  bulkQueryFetching: false,
  bulkBranchData: null,
  formData: null,
  uploadData: [],
});
