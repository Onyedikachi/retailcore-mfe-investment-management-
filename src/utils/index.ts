export * from "./capitalizeFirstLetter";
export * from "./capitalizeObjectKeys";
export * from "./context";
export * from "./getPercentage";
export * from "./removeNullFromObj";
export * from "./requestType";
export * from "./ucObjectKeys";
export * from "./handleDropdown";
export * from "./handleColorState";
export * from "./handleUserView";
export * from "./lowerObjectKeys";
