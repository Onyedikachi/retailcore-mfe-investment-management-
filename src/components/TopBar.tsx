import { useContext } from "react";
import { HiPlus } from "react-icons/hi";
import { Link } from "react-router-dom";
import Button from "./Button";
import { InvestmentContext, AppContext } from "../utils/context";
import { StatusCategoryType } from "@app/constants/enums";

export default function TopBar() {
  const { isChecker, category } = useContext(InvestmentContext);
  const { permissions } = useContext(AppContext);

  return (
    <div className="bg-white px-8 py-[52px] w-full border border-[#E5E9EB] min-h-[198px] flex gap-x-9 items-center border-t-0">
      <h1 className="text-[#636363] text-5xl font-bold">Branch Management</h1>

      {!isChecker &&
        category === StatusCategoryType.AllBranches &&
        permissions?.includes("CREATE_BRANCH") && (
          <Link to="/branch-management/create">
            <Button className="bg-sterling-red-800">
              <span className="p-[5px]">
                <HiPlus fontSize={14} />
              </span>{" "}
              Create new branch
            </Button>
          </Link>
        )}
    </div>
  );
}
