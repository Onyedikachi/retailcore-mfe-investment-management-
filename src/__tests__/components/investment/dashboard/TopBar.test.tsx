import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import TopBar from "../../../../components/branch-management/dashboard/TopBar";
import { MemoryRouter, Router } from "react-router-dom";
import { InvestmentContext } from "../../../../utils/context";
import { ProviderValue } from "../../../../__mocks__/fileMocks";
import { StatusCategoryType } from "../../../../constants/enums";

jest.mock("react-router-dom", () => ({
  BrowserRouter: ({ children }) => <div>{children}</div>,
  Routes: ({ children }) => <div>{children}</div>,
  Route: ({ element }) => element,
}));

describe("TopBar", () => {
  it("renders without crashing", () => {
    render(
      <InvestmentContext.Provider value={{ ...ProviderValue }}>
        <TopBar />
      </InvestmentContext.Provider>
    );
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <InvestmentContext.Provider value={{ ...ProviderValue }}>
        <TopBar />
      </InvestmentContext.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays heading", () => {
    const { getByText } = render(
      <InvestmentContext.Provider value={{ ...ProviderValue }}>
        <TopBar />
      </InvestmentContext.Provider>
    );

    expect(getByText("Branch Management")).toBeInTheDocument();
  });

  it("does not display button if isChecker is true", () => {
    render(
      <InvestmentContext.Provider value={{ ...ProviderValue }}>
        <TopBar />
      </InvestmentContext.Provider>
    );

    expect(screen.queryByText("Create new branch")).not.toBeInTheDocument();
  });

  // Tests that the 'Create new branch' button is not displayed when isChecker is true
  it("should not display the create new branch button when isChecker is true", () => {
    const { queryByText } = render(<TopBar />, {
      wrapper: ({ children }) => (
        <InvestmentContext.Provider
          value={{ isChecker: true, category: StatusCategoryType.AllBranches }}
        >
          {children}
        </InvestmentContext.Provider>
      ),
    });
    expect(queryByText("Create new branch")).not.toBeInTheDocument();
  });

  // Tests that the 'Create new branch' button is not displayed when category is not AllBranches
  it("should not display the create new branch button when category is not AllBranches", () => {
    const { queryByText } = render(<TopBar />, {
      wrapper: ({ children }) => (
        <InvestmentContext.Provider
          value={{ isChecker: false, category: StatusCategoryType.Requests }}
        >
          {children}
        </InvestmentContext.Provider>
      ),
    });
    expect(queryByText("Create new branch")).not.toBeInTheDocument();
  });
});
