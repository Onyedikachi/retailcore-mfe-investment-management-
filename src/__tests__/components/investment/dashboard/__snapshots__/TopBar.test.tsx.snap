// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white px-8 py-[52px] w-full border border-[#E5E9EB] min-h-[198px] flex gap-x-9 items-center border-t-0"
  >
    <h1
      class="text-[#636363] text-5xl font-bold"
    >
      Branch Management
    </h1>
  </div>
</DocumentFragment>
`;
